src/common/security/authcontext/authcontext.go

```diff
@@ -158,52 +152,51 @@ func Login(url, username, password string) (string, *AuthContext, error) {
 		Password: password,
 	})
 	if err != nil {
-		return "", nil, err
+		return nil, err
 	}
 
 	req, err := http.NewRequest(http.MethodPost, buildLoginURL(url), bytes.NewBuffer(data))
 	if err != nil {
-		return "", nil, err
-	}
-
-	code, header, data, err := send(req)
-	if code != http.StatusOK {
-		return "", nil, fmt.Errorf("failed to login with user %s: %d %s", username,
-			code, string(data))
-	}
-
-	ctx := &AuthContext{
-		Projects: make(map[string][]string),
-	}
-	if err = json.Unmarshal(data, ctx); err != nil {
-		return "", nil, err
+		return nil, err
 	}
 
-	return header.Get(AuthTokenHeader), ctx, nil
+	return send(client, req)
 }
 
-func send(req *http.Request) (int, http.Header, []byte, error) {
+func send(client *http.Client, req *http.Request) (*AuthContext, error) {
 	resp, err := client.Do(req)
 	if err != nil {
-		return 0, nil, nil, err
+		return nil, err
 	}
 	defer resp.Body.Close()
 
 	data, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return 0, nil, nil, err
+		return nil, err
 	}
-	return resp.StatusCode, resp.Header, data, nil
-}
 
-func buildCtxURL(url, principalID string) string {
-	url = strings.TrimRight(url, "/") + "/sso/auth-context"
-	if len(principalID) > 0 {
-		url += "/" + principalID
+	if resp.StatusCode != http.StatusOK {
```

Comment:
Need to double check for requests the expected response code is 200, otherwise let's put it as a parm.